test/llvmcall.jl

```diff
@@ -147,3 +147,34 @@ function confuse_declname_parsing()
     Void, Tuple{})
 end
 confuse_declname_parsing()
+
+
+module ObjLoadTest
+    using Base.Test
+    using Base.llvmcall
+    here = dirname(@__FILE__)
+    didcall = false
+    function callback()
+        global didcall
+        didcall = true
+        nothing
+    end
+    Base.ccallable(callback,Void,Tuple{},:jl_the_callback)
+    Base.ccallable(callback,Void,Tuple{},:_jl_the_callback)
+    # For now, only test this on Unix. On Windows we expect
+    # special structure in the object file to allow us to unwind,
+    # which this object file does not have.
+    @unix_only ccall(:jl_load_object_file,Void,(Ptr{UInt8},),joinpath(here,"objloadtest/objloadtest.o"))
+    function do_the_call()
+        llvmcall(
+        (""" declare void @call_the_callback_back()""",
+        """
+        call void @call_the_callback_back()
+        ret void
+        """),Void,Tuple{})
+    end
+    @unix_only begin
+	do_the_call()
+    	@test didcall
+    end
```

Comment:
Happy to have that test as well. It's not quite the same as shared libraries because it does see our JIT symbols, which is a crucial points. You (can)[https://github.com/Keno/DyldExport.jl] expose julia symbols to the dynamic linker, but it's even less pretty than this. The point of this functionality is to be able to simply load some native code into the running julia executable. It's not something I'd expect people to want to do often, but it can be very useful in limited cases.